services/analyze-sequelize-tables.js

```diff
@@ -62,100 +62,210 @@ function hasTimestamps(fields) {
+              manyToManyKeys.forEach((manyToManyKey) => {
+                databaseSchema[refTableName].references.push(setReference(foreignKey, 'belongsToMany', manyToManyKey));
+              });
+              isManyToMany = manyToManyKeys !== [];
+            }
+          });
+        }
+        if (!isManyToMany) {
+          const refPrimaryKeys = databaseSchema[refTableName].primaryKeys;
+          const refForeignKeys = databaseSchema[refTableName].foreignKeys;
 
-          references.push(reference);
-        } else if (type) {
-          // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
-          //         handle it automatically without necessary declaration.
-          if (!(nameColumn === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
-            const field = {
-              name: _.camelCase(nameColumn),
-              nameColumn,
-              type,
-              primaryKey: columnInfo.primaryKey,
-              defaultValue: columnInfo.defaultValue,
-            };
-
-            fields.push(field);
+          if (checkReferenceUnicity(refPrimaryKeys, refForeignKeys, refColumnName)) {
+            table.references.push(setReference(foreignKey, 'belongsTo'));
           }
+          databaseSchema[refTableName].references.push(
+            setReference(
+              foreignKey,
+              foreignKey.isPrimary || foreignKey.isUnique ? 'hasOne' : 'hasMany',
+            ),
+          );
         }
-      });
-
-      const options = {
-        underscored: isUnderscored(fields),
-        timestamps: hasTimestamps(fields),
-        hasIdColumn: hasIdColumn(fields, primaryKeys),
-        hasPrimaryKeys: !_.isEmpty(primaryKeys),
-      };
-
-      return {
-        fields,
-        references,
-        primaryKeys,
-        options,
-      };
+      }
     });
+  });
+}
+
+// NOTICE: Set the remaining fields
+async function analyzeTable(schema, foreignKeys, primaryKeys, references, table) {
+  const fields = [];
+  await P.each(Object.keys(schema), async (nameColumn) => {
+    const columnInfo = schema[nameColumn];
+    const type = await columnTypeGetter.perform(columnInfo, nameColumn, table);
+    const foreignKey = _.find(foreignKeys, { column_name: nameColumn, column_type: 'FOREIGN KEY' });
+
+    if (!(foreignKey
+          && foreignKey.foreign_table_name
+          && foreignKey.column_name && !columnInfo.primaryKey) && type) {
+      // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
+      //         handle it automatically without necessary declaration.
+      if (!(nameColumn === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
+        // NOTICE: Handle bit(1) to boolean conversion
+        if (columnInfo.defaultValue === "b'1'" || columnInfo.defaultValue === '((1))') { columnInfo.defaultValue = true; }
+        if (columnInfo.defaultValue === "b'1'" || columnInfo.defaultValue === '((0))') { columnInfo.defaultValue = false; }
```

Comment:
This make no sense `columnInfo.defaultValue === "b'1'"` this mean that columnInfo.defaultValue will always be false in this case 🤔 . Isn't it b'0' ?

Reply:
Check with @SteveBunlon

Reply:
My bad